store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedGraphStorage.java

```diff
@@ -598,23 +609,68 @@ private boolean changeGraphId(final String graphId, final String newGraphId, fin
                 }
             }
 
-            final GraphConfig configWithNewGraphId = new GraphConfig.Builder()
-                    .json(new GraphSerialisable.Builder().graph(graphToMove).build().getConfig())
-                    .graphId(newGraphId)
-                    .build();
+            //Update Tables
+            String storeClass = graphToMove.getStoreProperties().getStoreClass();
+            if (nonNull(storeClass) && storeClass.startsWith(AccumuloStore.class.getPackage().getName())) {
+                /*
+                 * uk.gov.gchq.gaffer.accumulostore.[AccumuloStore, SingleUseAccumuloStore,
+                 * SingleUseMockAccumuloStore, MockAccumuloStore, MiniAccumuloStore]
+                 */
+                try {
+                    AccumuloProperties tmpAccumuloProps = (AccumuloProperties) graphToMove.getStoreProperties();
+                    Connector connection = TableUtils.getConnector(tmpAccumuloProps.getInstance(),
+                            tmpAccumuloProps.getZookeepers(),
+                            tmpAccumuloProps.getUser(),
+                            tmpAccumuloProps.getPassword());
+
+                    if (connection.tableOperations().exists(graphId)) {
+                        connection.tableOperations().offline(graphId);
+                        connection.tableOperations().clone(graphId, newGraphId, true, null, null);
+                        connection.tableOperations().online(newGraphId);
+                        connection.tableOperations().delete(graphId);
+                    }
+                } catch (final Exception e) {
+                    LOGGER.warn("Error trying to update tables for graphID:{} graphToMove:{}", graphId, graphToMove);
+                    LOGGER.warn("Error trying to update tables.", e);
+                }
+            }
+
+            final GraphConfig configWithNewGraphId = cloneGraphConfigWithNewGraphId(newGraphId, graphToMove);
 
             //add the graph being renamed.
-            this.put(new GraphSerialisable.Builder()
+            GraphSerialisable newGraphSerialisable = new GraphSerialisable.Builder()
                     .graph(graphToMove)
                     .config(configWithNewGraphId)
-                    .build(), key);
+                    .build();
+            this.put(newGraphSerialisable, key);
+
+            //Update cache
+            if (isCacheEnabled()) {
+                try {
+                    federatedStoreCache.addGraphToCache(newGraphSerialisable, key, true/*true because graphLibrary should have throw error*/);
+                } catch (final CacheOperationException e) {
+                    //TODO FS recovery
```

Comment:
Is this still TODO?

Reply:
Kinda.
I think due to exceptions that is kinda hard logic to recover the graphs and trust that it was successfully recovered. Due to recovery could also throw exceptions.
I think meaningful Logging and get Admin to jump in and fix things is the best solution for recovery.

thoughts?

Reply:
Yeah I think throwing an exception explaining the cache broke is okay. I just think committing `//TODO FS recovery` to master isn't too descriptive

Reply:
The TODO wasn't for keeps. It was to prompt an examination.